piqueserver/extensions.py

```diff
@@ -0,0 +1,147 @@
+
+import os
+import importlib
+import imp
+
+from twisted.logger import Logger
+
+log = Logger()
+
+
+def check_scripts(script_names):
+    '''Validation for a list of regular extension scripts.
+
+    Check there are no duplicate scripts in the list of scripts names that is passed in.
+
+    Args:
+        script_names: The list of scripts names to be checked
+
+    Return:
+        bool: True if there are no duplicate scripts, False otherwise
+    '''
+    seen = set()
+    dups = []
+    for script in script_names:
+        if script in seen:
+            dups.append(script)
+        else:
+            seen.add(script)
+    if dups:
+        log.warn("Scripts included multiple times: {}".format(dups))
+        return False
+    return True
+
+
+def check_game_mode(game_mode_name):
```

Comment:
This function seems a bit too trivial... no real logic, single layer of abstraction, only used once. While making smaller functions is generally a good idea, it's important to not overdo it, as each function you add means you need to do one more "go to definition" while reading another function. This quickly adds up in larger codebases, and you end up spending most of your time jumping around checking the implementation of things.

Here's a longer article detailing the pros and cons: https://medium.com/@copyconstruct/small-functions-considered-harmful-91035d316c29